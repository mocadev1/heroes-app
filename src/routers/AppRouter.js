import React, { useContext } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
} from 'react-router-dom';

import PrivateRoutes from './PrivateRoutes';
import { AuthContext } from '../auth/AuthContext';

import LoginScreen from '../components/login/LoginScreen';
import DashboardRoutes from './DashboardRoutes';

const AppRouter = () => {

    const { user } = useContext(AuthContext);
    
    
    return (
        <Router>
            <div>

                {/* <Navbar /> // Eliminado ya que no requerimos el Navbar en el login */}

                <Switch>
                    <Route exact path="/login" component={ LoginScreen } />

                    <PrivateRoutes
                        path="/"
                        component={ DashboardRoutes } 
                        isAuthenticated={ user.logged }
                    />
                </Switch>

            </div>
        </Router>
    )
}

export default AppRouter
