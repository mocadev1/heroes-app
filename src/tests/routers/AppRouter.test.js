import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import AppRouter from '../../routers/AppRouter';

describe('<AppRouter />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }
    
    test('should display \'/login\' whether is not authenticated', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );
        

        console.log( wrapper.html() );
        expect( wrapper.find('h1').text() ).toContain('Login');
        expect( wrapper ).toMatchSnapshot();
    });

    test('should display <MarvelScreen /> whether is authenticated', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue } >
                <AppRouter />
            </AuthContext.Provider>
        );


        expect( wrapper.find('.navbar').exists() ).toBe(true);


    });
    
    
});
