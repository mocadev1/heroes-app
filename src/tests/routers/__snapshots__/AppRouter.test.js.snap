// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> should display '/login' whether is not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
